src/ServicePulse.Host/app/js/views/endpoint_details/endpoint_details.controller.js

```diff
@@ -4,41 +4,73 @@
     function controller(
         $scope,
         $routeParams,
+        $location,
         toastService,
         serviceControlService,
         monitoringService) {
 
         $scope.endpointName = $routeParams.endpointName;
         $scope.sourceIndex = $routeParams.sourceIndex;
         $scope.loading = true;
+        var subscription;
 
-        var subscription = monitoringService.endpointDetails($routeParams.endpointName, $routeParams.sourceIndex).subscribe(function (endpointInstances) {
-            if (endpointInstances.error) {
-                // show warning
-                toastService.showWarning('Could not load endpoint details');
-            } else {
-                $scope.endpointInstances = endpointInstances;
-                $scope.loading = false;
+        $scope.periods = [
+                { value: 5, text: "Last 5 min." },
+                { value: 10, text: "Last 10 min." },
+                { value: 15, text: "Last 15 min." },
+                { value: 30, text: "Last 30 min." },
+                { value: 60, text: "Last hour" }
+        ];
+
+        $scope.selectedPeriod = $scope.periods[0];
+
+        if ($location.$$search.historyPeriod) {
+            $scope.selectedPeriod = $scope.periods[$scope.periods.findIndex(function (period) {
+                return period.value == $location.$$search.historyPeriod;
+            })];
+        }
+
+        $scope.selectPeriod = function (period) {
+            $scope.selectedPeriod = period;
+
+            updateUI();
+        };
+
+        function updateUI() {
+            if (subscription) {
+                subscription.dispose();
             }
 
-            $scope.endpointInstances.forEach(function (instance) {
-                serviceControlService.getExceptionGroupsForEndpointInstance(instance.id).then(function (result) {
-                    instance.serviceControlId = result.data[0].id;
-                    instance.errorCount = result.data[0].count;
-                }, function (err) {
-                    // Warn user
+            subscription = monitoringService.createEndpointDetailsSource($routeParams.endpointName, $routeParams.sourceIndex, $scope.selectedPeriod.value).subscribe(function (endpointInstances) {
+                if (endpointInstances.error) {
+                    toastService.showWarning('Could not load endpoint details');
+                } else {
+                    $scope.endpointInstances = endpointInstances;
+                    $scope.loading = false;
+                }
+
+                $scope.endpointInstances.forEach(function (instance) {
```

Comment:
BTW. I know that we are not to concerned with performance of SP-SC communication but don't we want change API to make it a 1 call with an array of instanceID's?